.github/workflows/publish_packages.yml

```diff
@@ -0,0 +1,56 @@
+name: publish_packages.yml
+on:
+  release:
+    types:
+      - published
+  workflow_dispatch:
+
+jobs:
+  upload-to-registry:
+    name: Upload packages to the registry
+    runs-on: ubuntu-latest
+    env:
+      SCARB_REGISTRY_AUTH_TOKEN: ${{ secrets.SCARB_REGISTRY_AUTH_TOKEN }}
+    steps:
+      - uses: actions/checkout@v4
+
+      - uses: dtolnay/rust-toolchain@7b1c307e0dcbda6122208f10795a713336a9b35a
+        with:
+          toolchain: stable
+
+      - uses: software-mansion/setup-scarb@v1
+        with:
+          # todo: Use scarb stable version that support publishing plugins (after 2.8.4)
+          scarb-version: "nightly-2024-11-09"
+
+      - name: Check if package version exist in the registry
+        run: |
+          set -eo pipefail
+          snforge_scarb_plugin=$(curl -s https://scarbs.xyz/api/v1/index/sn/fo/snforge_scarb_plugin.json | jq --arg version "${{ steps.scarb-init.outputs.CAIRO_VERSION }}" '[.[] | select(.v == $version)] | length > 0')
+          snforge_std=$(curl -s https://scarbs.xyz/api/v1/index/sn/fo/snforge_std.json | jq --arg version "${{ steps.scarb-init.outputs.CAIRO_VERSION }}" '[.[] | select(.v == $version)] | length > 0')
+          sncast_std=$(curl -s https://scarbs.xyz/api/v1/index/sn/ca/sncast_std.json | jq --arg version "${{ steps.scarb-init.outputs.CAIRO_VERSION }}" '[.[] | select(.v == $version)] | length > 0')
+
+          echo "snforge_scarb_plugin=$snforge_scarb_plugin"
+          echo "snforge_std=$snforge_std"
+          echo "sncast_std=$sncast_std"
+
+          echo "snforge_scarb_plugin=$snforge_scarb_plugin" >> $GITHUB_OUTPUT
+          echo "snforge_std=$snforge_std" >> $GITHUB_OUTPUT
+          echo "sncast_std=$sncast_std" >> $GITHUB_OUTPUT
+
+      - name: Package and upload snforge_scarb_plugin
+        if: steps.check-plugins.outputs.snforge_scarb_plugin != 'true'
+        working-directory: crates/snforge-scarb-plugin
+        run: scarb publish
+
+      - name: Package and upload snforge_std
+        if: steps.check-plugins.outputs.snforge_std != 'true'
+        working-directory: snforge_std
+        run: |
+          ../scripts/set_plugin_version.sh
+          scarb publish --allow-dirty
```

Comment:
`--allow-dirty` because we modify `snforge_std` by executing the script without committing changes

Reply:
Why not require from tag that triggers the workflow to already include changes introduced by `set_plugin_version.sh`? 🤔

Reply:
For development reason, we always would like to have a relative path specified in the dependency

Reply:
Is it a hard requirement to set the version in snforge only for uploads? What are the downsides of having this set in repo?

Reply:
Yes, we would like always to test the latest plugin version while developing, so relative path has to be provided. e.g. e2e test
There is always a case where it should be changed this way or another, definitely here will be the easiest.